cmake/autodetect_executors.cmake

```diff
@@ -0,0 +1,21 @@
+set(GINKGO_HAS_OMP OFF)
+set(GINKGO_HAS_CUDA OFF)
+set(GINKGO_HAS_HIP OFF)
+find_package(OpenMP)
+include(CheckLanguage)
+check_language(CUDA)
+
+if(NOT DEFINED GINKGO_BUILD_OMP AND OpenMP_CXX_FOUND)
+    message(STATUS "Enabling OpenMP executor")
+    set(GINKGO_HAS_OMP ON)
+endif()
```

Comment:
```
if(OpenMP_CXX_FOUND)
    if(NOT DEFINED GINKGO_BUILD_OMP)
         message(STATUS "Enabling OpenMP executor")
    endif()
    set(GINKGO_HAS_OMP ON)
endif()
```
Maybe move check `NOT DEFINED GINKGO_BUILD_<LANG>` to make the GINKGO_HAS_<LANG> same every time

Reply:
Good idea, thanks!